language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java

```diff
@@ -345,6 +346,40 @@ public static boolean isUnionOfType(TypeSymbol typeSymbol, TypeDescKind typeDesc
                         .allMatch(kind -> kind == typeDescKind);
     }
 
+    /**
+     * Remove error types from the given type
+     *
+     * @param typeSymbol type descriptor to evaluate
+     * @param types instance of the Types
+     * @return {@link TypeSymbol} type symbol without error types
+     */
+    public static TypeSymbol removeErrorTypes(TypeSymbol typeSymbol, Types types) {
```

Comment:
Better if we can use Optional in cases where the method does not return a value, as it is a practice used in LS.

```suggestion
    public static Optional<TypeSymbol> removeErrorTypes(TypeSymbol typeSymbol, Types types) {
```

Reply:
changed